tests/integ/modin/window/test_expanding.py

```diff
@@ -174,3 +186,69 @@ def test_expanding_aggregation_series_unsupported(agg_func, agg_func_kwargs):
     snow_df = pd.Series([2, 3, 4, 1], index=["a", "b", "c", "d"])
     with pytest.raises(NotImplementedError):
         getattr(snow_df.expanding(), agg_func)(agg_func_kwargs)
+
+
+class TestTimedelta:
+    @pytest.mark.parametrize(
+        "create_function",
+        [
+            pytest.param(create_test_dfs, id="dataframe"),
+            pytest.param(create_test_series, id="series"),
+        ],
+    )
+    @pytest.mark.parametrize("agg_func", agg_func_not_supported_for_timedelta)
+    @sql_count_checker(query_count=0)
+    def test_expanding_aggregation_unsupported(self, create_function, agg_func):
+        eval_snowpark_pandas_result(
+            *create_function(
+                [
+                    native_pd.Timedelta(-1),
+                    native_pd.Timedelta(1),
+                    native_pd.Timedelta(2),
+                    native_pd.Timedelta(5),
+                    native_pd.NaT,
+                ]
+            ),
+            lambda object: getattr(object.expanding(), agg_func)(numeric_only=False),
+            expect_exception=True,
+            expect_exception_type=DataError,
+            # pandas error message is different for dataframe and series, but
+            # Snowpark pandas always uses the same message. Accept either
+            # message.
+            expect_exception_match=(
+                f"(?:{re.escape('No numeric types to aggregate')})|"
+                + f"(?:{re.escape(_TIMEDELTA_ROLLING_AGGREGATION_NOT_SUPPORTED)})"
```

Comment:
qq: do we need to re.escape those literals?

Reply:
We want to interpret the portions like `)|` and `(?:` as regex, not as literals within the regex, so we don't need to escape them.